agent/agent.go

```diff
@@ -603,14 +605,48 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 
 func (a *Agent) endTaskResponse(tc *taskContext, status string, message string) *apimodels.TaskEndDetail {
 	var description string
-	var cmdType string
+	var failureType string
+	if a.endTaskResp != nil { // if the user indicated a task response, use this instead
+		tc.logger.Task().Infof("Task status set using /task_status")
```

Comment:
nit: Use a constant for `task_status` and the endpoint definition or call it something more generic.